### PR TITLE
fix: Create entries for new reactive nodes in dependencyMap

### LINKED ISSUE #20401: [Bug]: Dependency is not being created for new reactive paths
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When a new property (for example newFun into existing js object) for an entity is getting created there is not an entry for it in the dependency map. 
Screenshot 2023-02-06 at 4 46 48 PM
Screenshot 2023-02-06 at 4 36 13 PM

check above dependency map screenshot doesn't have entry for `myFun1`

Another example of List widget v2
after adding `{{showAlert(currentView.Input1.text)}}` to list widget's button, current view is added(check differences below) but currentView entry is not available in dependency resulting showAlert won't have any value 

Screenshot 2023-02-06 at 4 58 36 PM
Screenshot 2023-02-06 at 4 51 44 PM
Screenshot 2023-02-06 at 4 52 12 PM

### Steps To Reproduce

1. Go to canvas & drop list widget v2
2. Drag and drop button widget inside list widget
3. add `{{showAlert(currentView.Input1.text)}}` on button click event
4. click button and showAlert will be empty

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ohansFavour**
### Cause of issue
When a reactive node was newly added (a possible change from `undefined` to a valid value), the `dependency map` was not updated. 

This PR :

- Fixes this issue by handling all possible cases of a NEW_EVENT
-  Cleans up the logic to update the dependency map

Fixes #20401 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Jest unit tests

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [x] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [x] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts [hunk 1]
```diff
@@ -1021,18 +1021,18 @@ class MetaWidgetGenerator {
...
+        Object.values(referencesEntityDef),
+      );
+    }
     /**
      * The above commented out code should be used instead of this
      * only after the following issue is resolved
      * https://github.com/appsmithorg/appsmith/issues/20401
      */
-    this.addCurrentViewProperty(metaWidget, Object.values(referencesEntityDef));
+    // this.addCurrentViewProperty(metaWidget, Object.values(referencesEntityDef));
```
Comment (Rishabh-Rathod) [inline app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts:1035]: Can we please also remove the commented code and also the above comment here?

Comment (ohansFavour) [reply app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts]: Done

**Review by ashit-rath — APPROVED** [review]
LGTM for the MetaWidgetGenerator.ts change

**Review by ApekshaBhosale — COMMENTED** [review]
Can you please add some comments in the newly created function that will give us clarity later and while debugging? @ohansFavour

**Comment by ramsaptami** [comment]
lgtm cc:@favour

File: app/client/src/workers/common/DependencyMap/index.ts [hunk 2]
```diff
@@ -177,100 +178,184 @@ export const updateDependencyMap = ({
...
+                });
+              }
+              if (isWidget(entity)) {
+                // update validation dependencies
+                dataTreeEvalRef.validationDependencyMap = {
+                  ...dataTreeEvalRef.validationDependencyMap,
+                  ...listValidationDependencies(entity, entityName),
+                };
+                didUpdateValidationDependencyMap = true;
+              }
```
Comment (Rishabh-Rathod) [inline app/client/src/workers/common/DependencyMap/index.ts:358]: This code is becoming more conditional and hard to understand. Can we discuss this once to make it more understandable?

**Review by Rishabh-Rathod — COMMENTED** [review]
The changes look good.

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

**Review by Rishabh-Rathod — APPROVED** [review]

**Pull request merged**